pilot/pkg/model/authorization.go

```diff
@@ -65,43 +65,80 @@ type AuthorizationPoliciesResult struct {
 	Audit  []AuthorizationPolicy
 }
 
+type ProxyInfo struct {
+	Namespace       string
+	WorkloadName    string
+	IsWaypointProxy bool
+	Workload        labels.Instance
+}
+
+// namespace string, workloadName string, isWaypointProxy bool, workload labels.Instance
 // ListAuthorizationPolicies returns authorization policies applied to the workload in the given namespace.
-func (policy *AuthorizationPolicies) ListAuthorizationPolicies(namespace string, workload labels.Instance) AuthorizationPoliciesResult {
+func (policy *AuthorizationPolicies) ListAuthorizationPolicies(proxyInfo ProxyInfo) AuthorizationPoliciesResult {
```

Comment:
Does `selectorAuthorizationPolicies` in pkg/serviceregistry/kube/controller/authorization.go also need to be updated to handle `targetRef`? If so, will this be addressed in a follow-up PR?

Reply:
Great find! I see that there is workload selector logic here. I need to understand how `selectorAuthorizationPolicies` is used before I can say we want to update it to handle targetRef for sure.

Reply:
From what I gather I believe we should update this since this is a function used to precompute **only** ambient workload information. 

What do you think? @keithmattix

Reply:
Yeah this is a great catch! Indeed, that function needs to be updated to handle targetRef as well

Reply:
Actually, these only deal with the ztunnel authz policies I think. So it may not need a change

Reply:
I wouldn't make a change to it in this PR. If needed in the future we can do a follow up PR.